Makefile

```diff
@@ -93,129 +92,49 @@ release: certs docker-release
 	@docker build -f scripts/dockerfiles/Dockerfile.release -t "amazon/amazon-ecs-agent:latest" .
 	@echo "Built Docker image \"amazon/amazon-ecs-agent:latest\""
 
-gogenerate:
-	go generate -x $(shell go list ./agent/...)
-	$(MAKE) goimports
-
 # We need to bundle certificates with our scratch-based container
 certs: misc/certs/ca-certificates.crt
 misc/certs/ca-certificates.crt:
 	docker build -t "amazon/amazon-ecs-agent-cert-source:make" misc/certs/
 	docker run "amazon/amazon-ecs-agent-cert-source:make" cat /etc/ssl/certs/ca-certificates.crt > misc/certs/ca-certificates.crt
 
-ifeq (${BUILD_PLATFORM},aarch64)
-test::
-	go test -tags unit -timeout=30s -v -cover $(shell go list ./agent/... | grep -v /vendor/)
-else
-test::
-	go test -race -tags unit -timeout=30s -v -cover $(shell go list ./agent/... | grep -v /vendor/)
-endif
-
-ifeq (${BUILD_PLATFORM},aarch64)
-test-silent::
-	go test -tags unit -timeout=30s -cover $(shell go list ./agent/... | grep -v /vendor/)
-else
-test-silent::
-	go test -race -tags unit -timeout=30s -cover $(shell go list ./agent/... | grep -v /vendor/)
-endif
-
-benchmark-test:
-	go test -run=XX -bench=. $(shell go list ./agent/... | grep -v /vendor/)
+gogenerate:
+	go generate -x ./agent/...
+	$(MAKE) goimports
 
-define dockerbuild
-	docker run \
-		--net none \
-		--user "$(USERID)" \
-		--volume "$(PWD)/out:/out" \
-		--volume "$(PWD):/go/src/github.com/aws/amazon-ecs-agent" \
-		--rm \
-		$(BUILDER_IMAGE) \
-		$(1)
-endef
+# 'go' may not be on the $PATH for sudo tests
+GO_EXECUTABLE=$(shell command -v go 2> /dev/null)
```

Comment:
does this work for non sudo as well?

Reply:
Yep. This just grabs the absolute path to the go binary (/usr/local/go/bin/go usually). 

If you install this via yum/apt, the root user will have go in its path. If you install it manually, root has no way to find the binary (and therefore needs the absolute path). 

Path considerations aside,  'go test' and '/usr/local/go/bin/go test' are equivalent.